src/OrchardCore/OrchardCore.Abstractions/Json/Dynamic/JsonDynamicValue.cs

```diff
@@ -1,35 +1,282 @@
+        ?? throw new InvalidCastException($"Cannot convert {value} to SByte");
+
+    public static explicit operator sbyte?(JsonDynamicValue value)
+        => value?.JsonValue?.GetValue<sbyte?>();
+
+    public static explicit operator float(JsonDynamicValue value)
+        => value?.JsonValue?.GetValue<float>()
+        ?? throw new InvalidCastException($"Cannot convert {value} to Float");
+
+    public static explicit operator float?(JsonDynamicValue value)
+        => value?.JsonValue?.GetValue<float?>();
+
+    public static explicit operator string?(JsonDynamicValue value)
+        => value?.JsonValue?.GetValue<string>();
+
+    public static explicit operator ushort(JsonDynamicValue value)
+        => value?.JsonValue?.GetValue<ushort>()
+        ?? throw new InvalidCastException($"Cannot convert {value} to UInt32");
+
+    public static explicit operator ushort?(JsonDynamicValue value)
+        => value?.JsonValue?.GetValue<ushort?>();
+
+    public static explicit operator uint(JsonDynamicValue value)
+        => value?.JsonValue?.GetValue<uint>()
+        ?? throw new InvalidCastException($"Cannot convert {value} to UInt32");
+
+    public static explicit operator uint?(JsonDynamicValue value)
+        => value?.JsonValue?.GetValue<uint?>();
+
+    public static explicit operator ulong(JsonDynamicValue value)
+        => value?.JsonValue?.GetValue<ulong>()
+        ?? throw new InvalidCastException($"Cannot convert {value} to UInt64");
+
+    public static explicit operator ulong?(JsonDynamicValue value)
+        => value?.JsonValue?.GetValue<ulong?>();
+
+    public static explicit operator byte[]?(JsonDynamicValue value)
+    {
+        if (value?.JsonValue.GetObjectValue() is string str)
+        {
+            return Convert.FromBase64String(str);
+        }
+
+        throw new InvalidCastException($"Cannot convert {value} to Byte array");
+    }
+
+    public static explicit operator TimeSpan(JsonDynamicValue value)
     {
-        get
+        if (value?.JsonValue?.GetObjectValue() is string str)
         {
-            if (!_hasValue)
+            return TimeSpan.Parse(str, CultureInfo.InvariantCulture);
+        }
+
+        throw new InvalidCastException($"Cannot convert {value} to TimeSpan");
+    }
+
+    public static explicit operator TimeSpan?(JsonDynamicValue value)
+    {
+        var str = value?.JsonValue?.GetObjectValue() as string;
+
+        return str is not null
+        ? TimeSpan.Parse(str, CultureInfo.InvariantCulture)
+        : null;
+    }
+
+    public static explicit operator Uri?(JsonDynamicValue value)
+        => new Uri(value?.JsonValue?.GetValue<string>() ?? string.Empty);
+
+    private sealed class JsonDynamicMetaObject : DynamicMetaObject
+    {
+        public JsonDynamicMetaObject(Expression expression, JsonDynamicValue value)
+            : base(expression, BindingRestrictions.Empty, value)
+        {
+        }
+
+        public override DynamicMetaObject BindConvert(ConvertBinder binder)
```

Comment:
What C# is invoking this code?

Reply:
I attempted to come up with a good comment 😇

Reply:
I don't think that makes sense, it's an overridden runtime method which is documented on the base class.